File: sistema-educativo-frontend/src/services/estudianteService.js
import axios from 'axios';

const API_URL = 'http://localhost:8080/api/estudiantes';

const estudianteService = {
    obtenerEstudiantes: ()  => axios.get(API_URL),
    obtenerEstudiantesPorId: (id) => axios.get(`${API_URL}/${id}`),
    crearEstudiante: (estudiante) => axios.post(API_URL, estudiante),
     actualizarEstudiante: (id, estudiante) => axios.put(`${API_URL}/${id}`, estudiante),
    eliminarEstudiante: (id) => axios.delete(`${API_URL}/${id}`)
};

export default estudianteService;

File: sistema-educativo/src/main/java/com/educacion/service/EstudianteService.java
package com.educacion.service;

import com.educacion.model.Estudiante;
import com.educacion.repository.EstudianteRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.ResponseEntity;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class EstudianteService {

    @Autowired
    private EstudianteRepository estudianteRepository;

    public List<Estudiante> obtenerEstudiantes() {
        return estudianteRepository.findAll();
    }

    public ResponseEntity<Estudiante> obtenerEstudiantePorId(String id) {
        return estudianteRepository.findById(id).map(ResponseEntity::ok)
                .orElse(ResponseEntity.notFound().build());
    }

    public Estudiante crearEstudiante(Estudiante estudiante) {
        return estudianteRepository.save(estudiante);
    }

    public ResponseEntity<Estudiante> actualizarEstudiante(String id, Estudiante estudiante) {
        return estudianteRepository.findById(id).map(e -> {
            e.setNombre(estudiante.getNombre());
            e.setCarrera(estudiante.getCarrera());
            return ResponseEntity.ok(estudianteRepository.save(e));
        }).orElse(ResponseEntity.notFound().build());
    }

    public ResponseEntity<Object> eliminarEstudiante(String id) {
        return estudianteRepository.findById(id).map(e -> {
            estudianteRepository.delete(e);
            return ResponseEntity.ok().build();
        }).orElse(ResponseEntity.notFound().build());
    }
}